### PR TITLE
Test Gifti types

### PR DESCRIPTION
**Pull request by effigies**
Candidate test + fixes.

There's an issue in Python 2 with 'Version' being `unicode` when parsed and a `str` in the default. Not sure what the best way to handle that is.

### CODE REVIEW

**Comment by effigies** [comment]
That's peculiar. Any idea why I'd see [this error](https://travis-ci.org/effigies/nibabel/jobs/150310367#L1358-L1359), given the check added in the latest commit? (That test passes on my machine.)

File: nibabel/gifti/parse_gifti_fast.py [hunk 1]
```diff
@@ -186,8 +186,8 @@ def StartElementHandler(self, name, attrs):
                 self.da.endian = gifti_endian_codes.code[attrs["Endian"]]
             if "ExternalFileName" in attrs:
                 self.da.ext_fname = attrs["ExternalFileName"]
-            if "ExternalFileOffset" in attrs:
-                self.da.ext_offset = attrs["ExternalFileOffset"]
+            if "ExternalFileOffset" in attrs and attrs["ExternalFileOffset"]:
```
Comment (matthew-brett) [inline nibabel/gifti/parse_gifti_fast.py]: Is this because `ExternalFileOffset` can be an empty string?

Comment (effigies) [inline nibabel/gifti/parse_gifti_fast.py]: Yes.

Comment (matthew-brett) [inline nibabel/gifti/parse_gifti_fast.py]: Worth making this explicit?  As in:

```
if "ExternalFileOffset" in attrs:
    self.da.ext_offset = attrs["ExternalFileOffset"] if attrs['ExternalFileOffset'] else 0
```

Comment (effigies) [inline nibabel/gifti/parse_gifti_fast.py]: Sure. Pushed.

**Pull request closed without merging**